src/decoder.cpp

```diff
@@ -96,146 +102,25 @@ void Decoder::main_loop() {
         }
     }
 
-    for (auto i = 0; i < bytes_read; i++) {
-        if (packed_) {
-            for (auto j = 0; j < 8; j++) {
-                this->process_bit((rx_buffer[i] >> j) & 0x1);
-            }
-        } else {
-            this->process_bit(rx_buffer[i]);
-        }
-    }
-}
-
-void Decoder::process_bit(uint8_t symbol) noexcept {
-    assert(symbol <= 1);
+    if (iq_or_bit_stream_) {
+        std::complex<float>* rx_buffer_complex = reinterpret_cast<std::complex<float>*>(rx_buffer);
```

Comment:
I think actually a static_cast is enough if u just want to change the pointer type.